### DISCUSSION TITLE
Various bugs when running in dev mode vs preview

### DISCUSSION THREAD

**Discussion by oysandvik94** [nuxt/nuxt#17925] [Questions]
Hi, I'm experiencing different bugs depending on what mode I'm running the nuxt3 server on, basically difference between production and dev. So if I run the server by just running the server with node: "node .output/server/index.mjs" or for example using the Azure Functions preset, I will encounter the bugs. If I run the server with "npx nuxi dev", "npx nuxi preview" or "npx nuxi start" the bugs don't appear.

**The bugs**
The bugs are mostly SSR related, but not all of them. I'm not looking to find solutions for all the bugs, since I'm more worried that I'm getting different results in dev vs prod, so overall that's the bigger issue. But to provide context and maybe hints to what's going on I'll try to describe them.

1. I'm have a component that looks essentially like this:
```
<template>
</tempalte>
<script lang="ts" setup>
import { useQuery } from '@urql/vue';

const result = useQuery({
            query: `
    query($take: Int!, $skip: Int!, $queryFilters: CompanyDtoFilterInput) {
        allCompanies{
            pageInfo {
            hasNextPage
            hasPreviousPage
            }
            totalCount
            items {
            id
            name
            }
        }
    }`,
           
        },
        );

        console.log(result);

definePageMeta({
  layout: "default",
});
</script>
```

When I run this in dev mode it will work fine. When I run it in "prod" mode I get this error:

> Error: use* functions may only be called during the `setup()` or other lifecycle hooks.
>     at useClient (file:///myProject/frontend/.output/server/node_modules/@urql/vue/dist/urql-vue.mjs:46:11)
>     at callUseMutation (file:///myProject/frontend/.output/server/node_modules/@urql/vue/dist/urql-vue.mjs:154:9)
>     at useMutation (file:///myProject/frontend/.output/server/node_modules/@urql/vue/dist/urql-vue.mjs:149:10)
>     at setup (file:///myProject/frontend/.output/server/chunks/server2.mjs:17259:5)
>     at _sfc_main$18.setup (file:///myProject/frontend/.output/server/chunks/server2.mjs:17349:26)
>     at callWithErrorHandling (file:///myProject/frontend/.output/server/chunks/index.mjs:1955:22)
>     at setupStatefulComponent (file:///myProject/frontend/.output/server/chunks/index.mjs:7503:29)
>     at setupComponent (file:///myProject/frontend/.output/server/chunks/index.mjs:7484:11)
>     at renderComponentVNode (file:///myProject/frontend/.output/server/chunks/index.mjs:9839:17)
>     at renderVNode (file:///myProject/frontend/.output/server/chunks/index.mjs:9951:22)
> [Vue warn]: Component  is missing template or render function.

This only happens on refresh, and not navigating inside the application after it has been rendered.

2. Inputs/props passed in to Components were undefined on the server side. So basically, when I tried to access them int he setup code the server would crash. This would not happen in dev mode. **This got fixed when I switched to using the script tag setup**.

3. When navigating from one specific page to another specific page (using router.push) I get this error:

> TypeError: Cannot read properties of null (reading 'insertBefore')
>     at insert (bootstrap-20153909.mjs:1:60129)
>     at d (bootstrap-20153909.mjs:1:40682)
>     at h (bootstrap-20153909.mjs:1:40311)
>     at hs.$ [as fn] (bootstrap-20153909.mjs:1:44778)
>     at hs.run (bootstrap-20153909.mjs:1:5034)
>     at ce (bootstrap-20153909.mjs:1:45044)
>     at ze (bootstrap-20153909.mjs:1:43851)
>     at re (bootstrap-20153909.mjs:1:43645)
>     at h (bootstrap-20153909.mjs:1:40434)
>     at hs.$ [as fn] (bootstrap-20153909.mjs:1:44778)

My nuxt config file looks like this:
```
import { defineNuxtConfig } from "nuxt3";

export default defineNuxtConfig({
  // Retrieve backend URl from env vars
  publicRuntimeConfig: {
    backendUrl: process.env.BACKEND_URL || 'https://localhost:5001/graphql',
    frontendUrl: process.env.FRONTEND_URL || 'http://localhost:3000'
  },

  css: ["@vueform/multiselect/themes/default.scss"],

  buildModules: [
    // pinia plugin - https://pinia.esm.dev
    ["@pinia/nuxt", "@nuxtjs/svg", "@vueuse/core/nuxt"],
  ],
  build: {
    postcss: {
      postcssOptions: {
        plugins: {
          tailwindcss: {},
          autoprefixer: {},
        },
      },
    },
  },

  vite: {
    ssr: {
      noExternal: ["@headlessui/vue"],
    },
  },
});
```

I'm on nuxt version 3.0.0-27383750.2b86345 currently. Latest node. 
The issues are happening on docker and multiple machines.

**Comment by dev-fredericfox** [comment]
> "3. When navigating from one specific page to another specific page (using router.push) I get this error:

I get this too on build. Is there a bug report filed already?

**Reply by Diizzayy** [reply]
can you provide a basic reproduction of this issue?
